### PR TITLE
Adding week 3,

### PR DESCRIPTION
**Pull request by kocheick**
first android app from scratch
I implemented circular photo, dark mode toggle and made sure they both work in landscape and portrait mode.

### CODE REVIEW

File: Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        ima
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt:12]: I'm not sure what this is. But that doesn't compile 😄

File: Week3/CardApp/app/src/main/res/layout-land/activity_main.xml [hunk 2]
```diff
@@ -0,0 +1,194 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/res/layout-land/activity_main.xml:10]: There's no need to have multiple ConstraintLayouts in this layout. You can build it all using just the root ConstraintLayout.

File: Week3/CardApp/app/src/main/res/layout-land/activity_main.xml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            android:layout_height="wrap_content"
+            android:paddingBottom="8dp"
+            android:text="Twitter / Snapchat : Kocheick"
+            android:textSize="13sp"
+            android:textStyle="bold"
+            app:layout_constraintStart_toStartOf="@+id/contact3"
+            app:layout_constraintTop_toBottomOf="@id/contact3" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+    <ScrollView
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/res/layout-land/activity_main.xml:134]: I'm not sure what you wanted to achieve with this ScrollView, but when I turn my phone to landscape, your layout breaks. The profile picture and a button on the bottom are cut off.

File: Week3/CardApp/app/src/main/res/layout-land/activity_main.xml [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/profilGroup"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
+        android:paddingTop="8dp"
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/res/layout-land/activity_main.xml:15]: We generally want to extract dimensions into a separate file called dimens.xml. 
That was also one of the requirements Jenn gave you for the homework.

File: Week3/CardApp/app/src/main/res/layout-land/activity_main.xml [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        app:layout_constraintBottom_toTopOf="@id/contactGroup"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <ImageView
+            android:id="@+id/imageView2"
+            android:layout_width="400px"
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/res/layout-land/activity_main.xml:21]: You don't want to use px in Android. We use `dp` for specifying dimensions and `sp` for text sizes.  By using `dp` views will have the same dimensions on different sized screens.

File: Week3/CardApp/app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+            app:layout_constraintStart_toStartOf="@+id/name"
+            app:layout_constraintTop_toBottomOf="@+id/job" />
+
+        <TextView
+            android:id="@+id/name"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="24dp"
+            android:text="@string/name"
+            android:textColor="#4E4E4E"
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/res/layout/activity_main.xml:54]: We also want to extract color codes to separate file, called colors.xml

File: Week3/CardApp/app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+                android:id="@+id/imageButton4"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_weight="1"
+                app:srcCompat="@android:drawable/ic_dialog_email" />
+        </LinearLayout>
+    </ScrollView>
+
+    <ImageButton
+        android:id="@+id/imageButton5"
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/res/layout/activity_main.xml:177]: Try to give your views meaningful ids. `imageButton5` doesn't really tell you much about what it does

File: Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt (hunk 1 above)
Comment (kocheick) [reply Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt:12]: Hey Lucas, I'm not really sure what happened here but there was definitely  more than just "ima". If you check the commit, you will see the last one was "to fix the screen orientation in dark mode.." but clearly I made a mistake while pushing the last commit.
Please, try again. I will update/revert this

File: Week3/CardApp/app/src/main/res/layout-land/activity_main.xml (hunk 3 above)
Comment (kocheick) [reply Week3/CardApp/app/src/main/res/layout-land/activity_main.xml:151]: I reuploaded the project for your review..

File: Week3/CardApp/app/src/main/res/layout-land/activity_main.xml (hunk 4 above)
Comment (kocheick) [reply Week3/CardApp/app/src/main/res/layout-land/activity_main.xml:16]: updated..

File: Week3/CardApp/app/src/main/res/layout-land/activity_main.xml (hunk 5 above)
Comment (kocheick) [reply Week3/CardApp/app/src/main/res/layout-land/activity_main.xml]: updated

File: Week3/CardApp/app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (kocheick) [reply Week3/CardApp/app/src/main/res/layout/activity_main.xml]: updated

File: Week3/CardApp/app/src/main/res/layout-land/activity_main.xml (hunk 2 above)
Comment (kocheick) [reply Week3/CardApp/app/src/main/res/layout-land/activity_main.xml:11]: I did that in order to group elements from same component together..  maybe there's a better way without using a constraint within another one ?

Comment (LukaKordic) [reply Week3/CardApp/app/src/main/res/layout-land/activity_main.xml:11]: There is a Group component in Constraint layout which you can use to group things together.

File: Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt [hunk 8]
```diff
@@ -1,15 +1,211 @@
...
         setContentView(R.layout.activity_main)
 
-        ima
+        loadUser(user)
+
+        showDialog()
+
+        darkModeSwitch.setOnClickListener {
+            switchTheme(darkModeSwitch.isChecked)
+
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt:39]: Remove blank line

File: Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt [hunk 9]
```diff
@@ -1,15 +1,211 @@
...
+        social.setOnClickListener {
+            val dialogTitle = social.text.toString()
+            val dialogMessage = "username: XBASS"
+
+            Builder(this).apply {
+                setTitle(dialogTitle)
+                setMessage(dialogMessage)
+                create()
+                show()
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt:80]: You can extract this into a separate method so you don't have to repeat this block of code every time you need to create a dialog.

File: Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt [hunk 10]
```diff
@@ -1,15 +1,211 @@
...
+            }
+        }
+
+
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt:84]: Blank line

File: Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt [hunk 11]
```diff
@@ -1,15 +1,211 @@
...
+        if (savedInstanceState != null) {
+            val isOn = savedInstanceState.getBoolean("isDarkModeOn")
+            switchTheme(isOn)
+
+
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt:89]: Remove blank lines

File: Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt [hunk 12]
```diff
@@ -1,15 +1,211 @@
...
+                view.setBackgroundColor(getColor(R.color.ligther_gray))
+            }
+            job.setTextColor(getColor(R.color.dark_red))
+            city.setTextColor(getColor(R.color.gray))
+            scheduleView.setTextColor(getColor(R.color.light_gray))
+            shareButtonView.setBackgroundColor(getColor(R.color.dirt_white))
+            this.darkModeSwitch.setTextColor(getColor(R.color.mainColor))
+
+
+        }
```
Comment (LukaKordic) [inline Week3/CardApp/app/src/main/java/com/example/cardapp/MainActivity.kt:176]: This is not a good way to implement dark mode. Imagine if you had really complex screen with 50 views on it. You would have to go through everything and change colors manually. 

Dark theme is implemented using themes & styles, read more about that here. https://developer.android.com/guide/topics/ui/look-and-feel/darktheme

**Pull request merged**